### ISSUE TITLE
setDataAtCell - remove [optional] marking for 2nd and 3rd paramater of the function

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/docs#181]
### Description
We define that the 2nd (column `index`), 3rd (`value`), and 4th (`source`) parameters of `setDataAtCell` are optional. 
![Zrzut ekranu 2020-10-9 o 14 38 32](https://user-images.githubusercontent.com/10757813/95584140-82e10580-0a3d-11eb-954d-b04bce20a508.png)
However, in fact, only the `source` is optional. When we set up the column and nothing more, not only we do not set anything but we also get a console error.

![Zrzut ekranu 2020-10-9 o 14 43 15](https://user-images.githubusercontent.com/10757813/95584314-cf2c4580-0a3d-11eb-8814-eaecd67adfe1.png)

Demo for v8.1.0 https://jsfiddle.net/AMBudnik/08ej5hxg/

The sibling method (used when data is an object) - `setDataAtRowProp` - is defined correctly (only the source is optional).

### Links

* setDataAtCell - https://handsontable.com/docs/Core.html#setDataAtCell
* setDataAtRowProp - https://handsontable.com/docs/8.1.0/Core.html#setDataAtRowProp

**Comment by scarletfog** [comment]
@AMBudnik as @wojciechczerniak said in https://github.com/handsontable/handsontable/pull/7346 that is, indeed, not true. When I looked closer into the docs I realized we can pass an array of arrays only and it's working correctly without succeeding params: ex. `hot.setDataAtCell([[5, 5, 'foo']]);`.

If you think the description lacks information maybe we could give the developer an example?

**Comment by AMBudnik** [comment]
@scarletfog That's true. Hm... however, if we check this example https://jsfiddle.net/AMBudnik/0kqp24ha/, where there's only the first (required) parameter, we get an error. Maybe adding some examples would remove all the confusion.

**Comment by aninde** [comment]
Version v8.3.0 did not bring any changes to this topic as connected PR was marked as a draft.
Demo v8.3.0 - same as in the above https://jsfiddle.net/aninde/5v94bych/ 

Example with array as the first and only argument, works https://jsfiddle.net/aninde/6Lkavo9r/
Example with first three parameters https://jsfiddle.net/aninde/ztcqo76u/

I think `setDataAtCell` works fine if we give 1 argument as array, it doesn't work when given as number.

I am withdrawing the issue to `In development` and I encourage to discussion. @wojciechczerniak WDYT?

**Comment by AMBudnik** [comment]
We get the same result in the new documentation https://handsontable.com/docs/api/core/#setdataatcell
